in_toto/in_toto_sign.py

```diff
@@ -0,0 +1,239 @@
+    [key_pub])
+  signable_object.verify_signatures(link_key_dict)
+
+def parse_args():
+  """
+  <Purpose>
+    A function which parses the user supplied arguments.
+
+  <Arguments>
+    None
+
+  <Exceptions>
+    None
+
+  <Returns>
+    Parsed arguments (args object)
+  """
+  parser = argparse.ArgumentParser(
+    description="in-toto-sign : Signs link file with/without replacement "
+                "of the existing signatures and dumps with/without "
+                "infixing the keyID in the filename")
+
+  lpad = (len(parser.prog) + 1) * " "
+
+  parser.usage = ("\n"
+                  "(sign | verify) --key <filepath>\n{0}"
+                  "[--replace-sig]\n{0}"
+                  "[--infix]\n{0}"
+                  "[--verbose]\n{0}"
+                  "<path/to/signable>\n\n"
+                  .format(lpad))
+
+  in_toto_args = parser.add_argument_group("in-toto-sign options")
+
+  in_toto_args.add_argument("operator", type=str, choices=['sign', 'verify'],
+                            help="Sign or Verify")
+
+  in_toto_args.add_argument("-k", "--key", type=str, required=True,
+                            help="Path to the key ")
+
+  in_toto_args.add_argument("-r", "--replace-sig", action="store_true",
+                            help="Replace all the old signatures, sign "
+                            "with the given key, and add the new "
+                            "signature in file")
+
+  in_toto_args.add_argument("-i", "--infix", action="store_true",
+                            help="Infix keyID in the filename while "
+                            "dumping, when -i the file will be dumped "
+                            "as original.<keyID>.link, "
+                            "else original.link")
+
+  in_toto_args.add_argument("-v", "--verbose", dest="verbose",
+                            help="Verbose execution.", default=False,
+                            action="store_true")
+
+  in_toto_args.add_argument("signablepath", type=str,
+                            help="path to the signable file")
+
+  args = parser.parse_args()
+  args.operator = args.operator.lower()
+
+  return args
+
+
+def main():
+  """
+  First calls parse_args to parse the arguments, and then calls either
+  add_sign or add_replace depending upon the arguments. Based on the
+  arguments it then dumps the corresponding file.
+  """
+  args = parse_args()
+  rsa_key = in_toto.util.import_rsa_key_from_file(args.key)
+
+  if args.verbose:
+    log.logging.getLogger().setLevel(log.logging.INFO)
+
+  try:
+    if args.operator == 'sign':
+      if args.replace_sig:
+        signable_object = replace_sign(args.signablepath, args.key)
+
+      else:
+        signable_object = add_sign(args.signablepath, args.key)
+
+      if args.infix:
+        signable_object.dump(key=rsa_key)
+        sys.exit(0)
+
+      else:
+        signable_object.dump()
+        sys.exit(0)
+
+    else:
+      verify_sign(args.signablepath, args.key)
```

Comment:
Could you put this in a separate `try/except` block (not inside this try) and log with `log.fail_verification` on `SignatureVerificationError`, and `log.error` on all other exceptions?

Maybe it would also be nice to log with `log.pass_verification` if the verification passes?